src/nvim/fileio.c

```diff
@@ -1423,29 +1423,36 @@ readfile (
               }
             }
           }
-          if (enc_utf8) {               /* produce UTF-8 */
+          if (enc_utf8) {               // produce UTF-8
             dest -= utf_char2len(u8c);
-            (void)utf_char2bytes(u8c, dest);
-          } else {                    /* produce Latin1 */
-            --dest;
+            assert(u8c <= INT_MAX);
```

Comment:
put it before `utf_char2len`, which needs the same cast.

Reply:
done